sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1237,9 +1239,11 @@ template <typename T, int NElts, lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           typename AccessorTy,
           typename FlagsT = __ESIMD_DNS::dqword_element_aligned_tag>
-__ESIMD_API std::enable_if_t<!std::is_pointer_v<AccessorTy> &&
-                                 __ESIMD_NS::is_simd_flag_type_v<FlagsT>,
-                             __ESIMD_NS::simd<T, NElts>>
+__ESIMD_API std::enable_if_t<
+    __ESIMD_DNS::is_accessor_with_v<AccessorTy,
```

Comment:
this probably should check not for any accessor, but for device_accessor, right?
Local accessor always accept 32-bit offset, while this prototype accepts 64-bit offset in stateless mode. (See `__ESIMD_DNS::DeviceAccessorOffsetT offset` below)